stdlib/private/StdlibCollectionUnittest/CheckMutableCollectionType.swift.gyb

```diff
@@ -907,16 +907,20 @@ func checkPartition_${'Predicate' if predicate else 'WhereElementIsComparable'}(
 
   var c = makeWrappedCollectionWithComparableElement(elements)
 %   end
-
+  let pivotElt  = c.first
+  var pivot = c.startIndex
 %   if predicate:
   let closureLifetimeTracker = LifetimeTracked(0)
-  let pivot = c.partition() {
-    (lhs, rhs) in
-    _blackHole(closureLifetimeTracker)
-    return extract(lhs).value < extract(rhs).value
+  if let first = pivotElt {
+    pivot = c.partition(by: {
```

Comment:
Thanks @gribozavr! I've modified this test to take a pivot value rather than pivoting on the first element. This way we can also test the case where no elements in the collection pass the predicate and every element in the collection passes. Given that `partition(by:)` doesn't care about the Comparable-ness of the elements, I'm not sure how important the Opaque wrapper is, but I've left it in so we could use the identity-checking mechanism.